src/Shared/ProjectWriter.cs

```diff
@@ -166,12 +165,25 @@ public override void WriteString(string text)
             }
         }
 
+        /// <summary>
+        /// Override method in order to omit the xml declaration tag in certain cases. The tag will be written if:
+        ///  - The tag was present in the file/stream loaded.
+        ///  - The Encoding is specified and not default (UTF8)
+        /// </summary>
+        public override void WriteStartDocument()
+        {
+            if (_writeXmlDeclaration)
+            {
+                base.WriteStartDocument();
```

Comment:
It does not get stripped. In testing just now with VS, the BOM will be added when the encoding is UTF8. But it no longer (even without project system changes) adds the <xml> tag.

---
In reply to: [80529908](https://github.com/Microsoft/msbuild/pull/1089#discussion_r80529908) [](ancestors = 80529908)